app/grandchallenge/workstation_configs/views.py

```diff
@@ -11,3 +26,52 @@ class WorkstationConfigViewSet(ReadOnlyModelViewSet):
     serializer_class = WorkstationConfigSerializer
     queryset = WorkstationConfig.objects.all()
     permission_classes = [IsAuthenticated]  # Note: this is a ReadOnlyView
+
+
+class WorkstationConfigList(LoginRequiredMixin, ListView):
+    model = WorkstationConfig
+
+
+class WorkstationConfigCreate(
+    LoginRequiredMixin, PermissionRequiredMixin, CreateView
+):
+    model = WorkstationConfig
+    form_class = WorkstationConfigForm
+    permission_required = f"{WorkstationConfig._meta.app_label}.add_{WorkstationConfig._meta.model_name}"
+
+    def form_valid(self, form):
+        form.instance.creator = self.request.user
+        return super().form_valid(form)
+
+
+class WorkstationConfigDetail(LoginRequiredMixin, DetailView):
+    model = WorkstationConfig
+
+
+class WorkstationConfigUpdate(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, UpdateView
+):
+    model = WorkstationConfig
+    form_class = WorkstationConfigForm
+    permission_required = f"change_{WorkstationConfig._meta.model_name}"
```

Comment:
Question: is the app label required? eg,
```suggestion
    permission_required = f"{WorkstationConfig._meta.app_label}.change_{WorkstationConfig._meta.model_name}"
```

Reply:
I think it also works without it, but it is of course cleaner to include it.